src/shims/unix/fs.rs

```diff
@@ -940,13 +939,16 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     }
 
     // `lstat` is used to get symlink metadata.
-    fn macos_lstat(
+    fn macos_fbsd_lstat(
         &mut self,
         path_op: &OpTy<'tcx, Provenance>,
         buf_op: &OpTy<'tcx, Provenance>,
     ) -> InterpResult<'tcx, Scalar<Provenance>> {
         let this = self.eval_context_mut();
-        this.assert_target_os("macos", "lstat");
+
+        if !matches!(&*this.tcx.sess.target.os, "macos" | "freebsd") {
+            panic!("macos_fbsd_lstat not meant for os {}", this.tcx.sess.target.os);
```

Comment:
There's a verb missing in this sentence.
```suggestion
            panic!("`macos_fbsd_lstat` should not be called on {}", this.tcx.sess.target.os);
```
(Do this everywhere you have the new panic.)